overlays/nodejs/lambda/lambdaMixins.ts

```diff
@@ -0,0 +1,308 @@
+     * once.  The factory-function will be invoked once when the final AWS Lambda module is loaded.
+     * It can run whatever code it needs, and will end by returning the actual function that Lambda
+     * will call into each time the Lambda is invoked.
+     */
+    factoryFunc?: EntryPointFactory<E, R>;
+
+    /**
+     * A pre-created role to use for the Function. If not provided, [policies] will be used.
+     */
+    role?: iam.Role;
+
+    /**
+     * A list of IAM policy ARNs to attach to the Function.  Will be used if [role] is not provide.
+     * If neither [role] nor [policies] is provided, a default policy of [iam.AWSLambdaFullAccess]
+     * will be used instead.
+     */
+    policies?: arn.ARN[];
+
+    /**
+     * The Lambda runtime to use.  If not provided, will default to [NodeJS8d10Runtime]
+     */
+    runtime?: runtime.Runtime;
+
+    /**
+     * The paths relative to the program folder to include in the Lambda upload.  Default is `[]`.
+     */
+    includePaths?: string[];
+
+    /**
+     * The packages relative to the program folder to include in the Lambda upload.  The version of
+     * the package installed in the program folder and it's dependencies will all be included.
+     * Default is `[]`.
+     */
+    includePackages?: string[];
+
+    /**
+     * The packages relative to the program folder to not include the Lambda upload. This can be
+     * used to override the default serialization logic that includes all packages referenced by
+     * project.json (except @pulumi packages).  Default is `[]`.
+     */
+    excludePackages?: string[];
+
+    /**
+     * @internal.  TODO(cyrusn): Remove if not needed by higher layers.
+     */
+    serialize?: (obj: any) => boolean
+}>;
+
+/**
+ * Base type for all subscription types.  An event subscription represents a connection between some
+ * AWS resource an an AWS lambda that will be triggered when something happens to that resource.
+ */
+export class EventSubscription extends pulumi.ComponentResource {
+    public permission: permission.Permission;
+    public func: lambdaFunction.Function;
+
+    public constructor(
+        type: string, name: string, func: lambdaFunction.Function, props: Record<string, any>, opts?: pulumi.ResourceOptions) {
+
+        super(type, name, props, opts);
+
+        this.func = func;
+    }
+}
+
+/**
+ * Creates an AWS Lambda from a provided Handler.  If the Handler is already an AWS Lambda, this
+ * does nothing and simply returns that instance.  Otherwise, if it is a JavaScript Function
+ * instance, this will serialize that Function instance into a usable packaged form that is then
+ * exposed through an AWS Lambda.
+ *
+ * If no [policies] are included in [functionOptions] a default policy of
+ * [aws.iam.AWSLambdaFullAccess] will be used.
+ */
+export function createLambdaFunction<E, R>(
+    name: string, options: FunctionOptions<E, R>, opts?: pulumi.ResourceOptions): lambdaFunction.Function {
```

Comment:
Note: this majority of this code was pulled out of aws.serverless.Function.  That type now just forwards to this helper.